Assets/Scripts/UI/CommandLine/EMCli.cs

```diff
@@ -3,62 +3,171 @@
+
+    private static bool autofocus => focusOnEnable && FocusOnEnable; //First one for technical realization and second one for settings
+    private static bool focusedOnInputLine => GUI.GetNameOfFocusedControl().Equals(nameOfControle);
 
     // Use this for initialization
     void Start()
     {
+        UnityEngine.Object.DontDestroyOnLoad(base.gameObject);
+        Messages = new List<ConsoleMessage>();
         EnterCommand("/info");
     }
 
     // Update is called once per frame
     void Update()
     {
-        //if (Input.GetKeyDown(KeyCode.F3)) //Does not work here
-        //{
-        //    Visible = !Visible;
-        //}
-        //else if (Input.GetKeyDown(KeyCode.UpArrow))
-        //{
-        //    InputLine.Up();
-        //}
-        //else if (Input.GetKeyDown(KeyCode.DownArrow))
-        //{
-        //    InputLine.Down();
-        //}
+
+    }
+
+    //private static bool showOrHide(KeyCode keyCode)
+    //{
+    //    return keyCode == (keyCodeShowHide | keyCodeShowHide2);
+    //}
+
+    public static void SetupGUI(float width, float height)
+    {
+        LayoutWidth = width;
+        LayoutHeight = height;
+        rectMainGUI = new Rect(0, 0, LayoutWidth, LayoutHeight);
+        guiLayoutOptionMainGUI = new GUILayoutOption[] { GUILayout.Width(LayoutWidth), GUILayout.MaxHeight(LayoutHeight) };
+    }
+
+    public static void ResetGUISize()
+    {
+        SetupGUI(320, 430);
+    }
+
+    public static void SwitchBack()
+    {
+        WindowFunctionMainGUI = new GUI.WindowFunction(WindowLayoutMainGUI);
+    }
+
+    public static void SwitchDebugLevel(DebugLevel level)
+    {
+        foreach(DebugLevelOption dloption in DebugLevelOption.DebugLevelOptions)
+        {
+            if (level.Equals(dloption.DebugLevel))
+            {
+                dloption.Enabled = !dloption.Enabled;
+                break;
+            }
+        }
+    }
+
+    public static void ExcludeLevel(DebugLevel level)
+    {
+        foreach (DebugLevelOption dloption in DebugLevelOption.DebugLevelOptions)
+        {
+            if (level.Equals(dloption.DebugLevel))
+            {
+                dloption.Enabled = false;
+                break;
+            }
+        }
+    }
+
+    public static void IncludeLevel(DebugLevel level)
+    {
+        foreach (DebugLevelOption dloption in DebugLevelOption.DebugLevelOptions)
+        {
+            if (level.Equals(dloption.DebugLevel))
+            {
+                dloption.Enabled = true;
+                break;
+            }
+        }
     }
 
-    public static void ClearLayout()
+    public static void RefreshLayout()
     {
         layout = string.Empty;
+        foreach(ConsoleMessage message in Messages)
+        {
+            foreach(DebugLevelOption option in DebugLevelOption.DebugLevelOptions)
+            {
+                if (option.DebugLevel.Equals(message.ConsoleMessageDebugLevel))
```

Comment:
Why a double if statement here? Just do
```suggestion
                if (option.DebugLevel.Equals(message.ConsoleMessageDebugLevel) && option.Enabled)
```